### PR TITLE
Move SnarkyJS bindings to separate repo

### PR DESCRIPTION
**Pull request by mitschabaude**
SnarkyJS companion: https://github.com/o1-labs/snarkyjs/pull/822
New bindings repo: https://github.com/o1-labs/snarkyjs-bindings (located as a nested git submodule inside snarkyjs; checked out at the current `main` branch)

Changes to the Mina repo:
* The kimchi JS/Wasm bindings, previously located at `src/lib/crypto/kimchi_bindings/{wasm,js}`, is moved down into `snarkyjs-bindings/kimchi`
* The OCaml parts of snarkyjs, previously at `src/lib/snarky_js_bindings`, are moved down into `snarkyjs-bindings/ocaml`.
* Some snarkyjs tests in `../snarky_js_bindings/test_module` are moved down into snarkyjs
* Other snarkyjs tests in `../snarky_js_bindings/tests` are deleted, because they didn't test anything not covered by other unit tests
* The `src/lib/snarky_js_bindings` folder, which at this point would have only contained `/snarkyjs`, is deleted; the snarkyjs submodule is moved up to `src/lib/snarkyjs`
* snarkyjs-related scripts inside `/scripts` are moved down into `snarkyjs-bindings/scripts`; old script locations remain and now call scripts at the new location. Motivation for this was to have scripts located next to their build targets, so they can be changed together in the same repo 
* Remove `mina_signer` make target and mina-signer tests from CI (these were targeting the old, no longer used mina-signer based on the `client_sdk`)
* Update some file paths and CODEOWNERS

### CODE REVIEW

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-build-me

File: .gitmodules [hunk 1]
```diff
@@ -18,5 +18,5 @@
 	path = src/external/c-reference-signer
 	url = https://github.com/MinaProtocol/c-reference-signer.git
 [submodule "src/lib/snarky_js_bindings/snarkyjs"]
```
Comment (Trivo25) [inline .gitmodules:20]: do we not need to rename this part?

Comment (mitschabaude) [reply .gitmodules:20]: I don't think so 😅

**Review by Trivo25 — APPROVED** [review]

**Review by mrmr1993 — APPROVED** [review]

**Review by bkase — APPROVED** [review]

**Comment by mitschabaude** [comment]
!ci-build-me

**Pull request merged**